src/ripple/net/impl/RPCCall.cpp

```diff
@@ -589,65 +589,63 @@ class RPCParser
         return parseAccountRaw2 (jvParams, jss::destination_account);
     }
 
-    // channel_authorize <private_key> <channel_id> <drops>
+    // channel_authorize <private_key> <json>
     Json::Value parseChannelAuthorize (Json::Value const& jvParams)
     {
-        Json::Value jvRequest (Json::objectValue);
+        Json::Value  jvClaim;
+        Json::Reader reader;
 
-        jvRequest[jss::secret] = jvParams[0u];
+        if (reader.parse (jvParams[1u].asString (), jvClaim))
         {
-            // verify the channel id is a valid 256 bit number
-            uint256 channelId;
-            if (!channelId.SetHexExact (jvParams[1u].asString ()))
-                return rpcError (rpcCHANNEL_MALFORMED);
-        }
-        jvRequest[jss::channel_id] = jvParams[1u].asString ();
+            try
+            {
+                auto const drops = std::stoul (
+                    jvClaim[jss::Amount].asString ());
+                (void)drops;  // just used for error checking
+            }
+            catch (std::exception const&)
+            {
+                return rpcError (rpcCHANNEL_AMT_MALFORMED);
+            }
 
-        try
-        {
-            auto const drops = std::stoul (jvParams[2u].asString ());
-            (void)drops;  // just used for error checking
-            jvRequest[jss::amount] = jvParams[2u];
-        }
-        catch (std::exception const&)
-        {
-            return rpcError (rpcCHANNEL_AMT_MALFORMED);
+            Json::Value jvRequest;
+            jvRequest[jss::secret] = jvParams[0u].asString ();
+            jvRequest[jss::tx_json] = jvClaim;
+
+            return jvRequest;
         }
 
-        return jvRequest;
+        return rpcError (rpcINVALID_PARAMS);
     }
 
-    // channel_verify <public_key> <channel_id> <drops> <signature>
+    // channel_verify <json>
     Json::Value parseChannelVerify (Json::Value const& jvParams)
     {
-        std::string const strPk = jvParams[0u].asString ();
-
-        if (!parseBase58<PublicKey> (TokenType::TOKEN_ACCOUNT_PUBLIC, strPk))
-            return rpcError (rpcPUBLIC_MALFORMED);
+        if (1 == jvParams.size ())
+        {
+            Json::Value  jvClaim;
+            Json::Reader reader;
+            if (reader.parse (jvParams[0u].asString (), jvClaim))
+            {
+                try
+                {
+                    auto const drops = std::stoul (
```

Comment:
Not sure how I feel about `std::stoul`. Maybe consider `beast::lexicalCastThrow` instead?